nvflare/fuel/utils/config.py

```diff
@@ -0,0 +1,149 @@
+        """
+            Return the original underline config object representation if you prefer to use it directly
+            Pyhocon → ConfigTree
+            JSON → Dict
+            OMEGACONF → ConfigDict
+
+        Returns: Any,
+            return native config objects
+        """
+
+        pass
+
+    @abstractmethod
+    def get_format(self):
+        """
+            returns the current config objects ConfigFormat
+        Returns:
+            return ConfigFormat
+        """
+
+        pass
+
+    @abstractmethod
+    def get_location(self) -> Optional[str]:
+        """
+           return the file path where this configuration is loaded from
+
+        Returns:
+            return None if the config is not from file else return file path
+
+        """
+        pass
+
+    @abstractmethod
+    def to_dict(self) -> Dict:
+        """
+            convert underline config object to dictionary
+
+        Returns:
+            Returns: converted configuration as dict
+
+        """
+        pass
+
+    @abstractmethod
+    def to_conf_str(self, element: Dict) -> str:
+        """
+            convert dict element to the str representation of the underline configuration.
+            For example, for JsonFormat, the method return json string
+            for PyhoconFormat, the method return pyhocon string
+            for OmegaconfFormat, the method returns YAML string representation
+
+        Args:
+            element: dict
+
+        Returns:
+            string representation of the configuration in given format
+
+        """
+        pass
+
+
+class ConfigLoader(ABC):
+    @abstractmethod
+    def load_config(
+        self, file_path: str, default_file_path: Optional[str] = None, overwrite_config: Optional[Dict] = None
+    ) -> Config:
+        """
+            configuration from default_file_path will be the default config if specified
+            configuration from file_path will be the merge with default config overwrite the same key
+            configuration from overwrite_config if provided will be the merge with config overwrite the same key
+
+        Args:
+            file_path: file path for configuration to be loaded
+            default_file_path: file path for default configuration
+            overwrite_config: dict config that will  overwrite the final config if provided
+
+        Returns:
+            return Config
+        """
+
+        pass
+
+    def load_config_from_str(self, config_str: str) -> Config:
+        """
+            Load Configuration based on the string representation of the underline configuration
+            for example, Json String for Jsonformat. python conf string or yaml string presentation
+
+        Args:
+            config_str:
+        Returns:
+            return Config
+
+        """
+        raise NotImplementedError
+
+    def load_config_from_dict(self, config_dict: dict) -> Config:
+        """
+            Load Configuration based for given config dict.
```

Comment:
Instead of starting in the next line, it should starts right after """.

For example:  `"""Loads Configuration based for given config dict.`